linkedlist/linkedList.go

```diff
@@ -108,3 +139,46 @@ func (l *LinkedList) HandleList() []int {
 	}
 	return values
 }
+
+func (l *LinkedList) SearchConcurrently(ctx context.Context, cancel context.CancelFunc, find int) (int, bool) {
+	var result int
+	var isFound bool
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+
+	for i := 0; i < len(Nodes); i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			index := Nodes[i]
+			for j := 0; j < int(part); j++ {
+				select {
+				case <-ctx.Done():
+					return
+				default:
+				}
+
+				if index == nil {
+					return
+				}
+
+				if index.Value == find {
+					mu.Lock()
+					result = j + i
+					isFound = true
+					mu.Unlock()
+					cancel()
+					return
+				}
```

Comment:
I think we can use sync.Once in this section
it ensure that  cancel function is executed only once even if multiple goroutines try to call it this is useful when multiple goroutines may find the value at the same time and they may call cancel func
@armineyvazi @alipourhabibi 
```
var cancelCalled sync.Once

if index.Value == find {
	mu.Lock()
	if !isFound {
	    result = offset + j
            isFound = true
            cancelCalled.Do(cancel) 
	}
	mu.Unlock()
	return
}
```